modules/renter/hostdb/hostdb.go

```diff
@@ -44,11 +44,13 @@ type HostDB struct {
 	// the scanPool is a set of hosts that need to be scanned. There are a
 	// handful of goroutines constantly waiting on the channel for hosts to
 	// scan. The scan map is used to prevent duplicates from entering the scan
-	// pool.
+	// pool. The scanWG is used to wait for the last generation of scanning
+	// threads to stop before creating new ones.
 	scanList        []modules.HostDBEntry
 	scanMap         map[string]struct{}
 	scanWait        bool
 	scanningThreads int
+	scanWG          *sync.WaitGroup
```

Comment:
hmm. I would prefer to keep this out of the hostdb object if possible. The less shared state, the better. I was also under the impression that waitgroups couldn't be reused, but apparently that is not true. Still, it feels like `scanWait` and `scanWG` overlap in responsibility somewhat. If `scanWait` can tell you whether there's a scan running, why do we need to call `scanWG.Wait` at all?

Reply:
`scanWait` tells you if the "main thread" is running. The one that spawns scanning threads. Setting `scanWait` is therefore required to avoid creating a new "main thread" and requires the lock to be held. Unfortunately you can't hold the lock while waiting on `scanWG`. 
The issue is that you can't unlock the lock to wait on `scanWG` after checking if `scanList` is empty. If you do, there is a chance that new scan jobs are added to the list while we wait for the threads to finish. Those jobs are not executed and therefore cause some tests to randomly fail.